src/wallet/bdb.cpp

```diff
@@ -528,17 +528,17 @@ bool BerkeleyBatch::Rewrite(BerkeleyDatabase& database, const char* pszSkip)
                         fSuccess = false;
                     }
 
-                    Dbc* pcursor = db.GetCursor();
-                    if (pcursor)
+                    if (db.CreateCursor())
```

Comment:
In commit "walletdb: Handle cursor internally" (84bc5c1705691f4ecdfb8bcc90eb1bc4970281dc)

As noted https://github.com/bitcoin/bitcoin/pull/18971#discussion_r437625076, the error handling for failing to get a cursor here is broken. It had been broken before this commit, but now this commit is introducing a new way for getting a cursor to fail (the case where m_cursor is not null), so this might be making the problem worse. A simple way to fix this while keeping status quo would be to assert(!m_cursor) in CreateCursor instead of returning false. Otherwise it probably would be good to do improve the error handling here and at least log or return failure.

Reply:
I've changed it to assert.